streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -728,6 +728,7 @@
     public static class InternalConfig {
         public static final String TASK_MANAGER_FOR_PARTITION_ASSIGNOR = "__task.manager.instance__";
         public static final String VERSION_PROBING_FLAG = "__version.probing.flag__";
+        public static final String SHUTDOWN_ERROR_CODE = "__shutdown.error.code__";
```

Comment:
Version probing is a boolean flag.
Assignment error code is int.
If we unify these two, we need to encode version probing. Since the range for assignment error code is open ended, it may not be obvious how we encode version probing.

BTW version probing doesn't imply assignment error.

Reply:
Yes, my intention is to use another error code value (seems you've already done it as `VERSION_PROBING`) to replace the flag, and then after onAssignment is called we would check the error code and if it is not `NONE` react accordingly, it may be simply shutdown and stop-the-world, or other actions like down-grade.